plugins/filter/lists.py

```diff
@@ -0,0 +1,118 @@
+# -*- coding: utf-8 -*-
+# Copyright (c) Ansible Project
+# GNU General Public License v3.0+ (see LICENSES/GPL-3.0-or-later.txt or https://www.gnu.org/licenses/gpl-3.0.txt)
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+from ansible.errors import AnsibleFilterError
+from ansible.module_utils.common.collections import is_sequence
+
+
+def remove_duplicates(lst):
+    seen = set()
+    result = []
+    for item in lst:
+        if item not in seen:
+            seen.add(item)
+            result.append(item)
+    return result
+
+
+def flatten(lst):
```

Comment:
One thing I don't like about flatten is that you cannot get rid of it. For example, with the ansible-core `union` filter, you can do `[['a']] | union([['b'], ['a']])` to get `[['a'], ['b']]`. With your `community.general.lists_union` filter, you get `['a', 'b']` due to flattening.

How about making flattening configurable with a `flatten` keyword argument (with whichever default you prefer for it)?

(Actually your code wouldn't work with lists anyway - but it wouldn't work with other complex types like dicts either, see my next comment.)